### PR TITLE
Check role privileges when validating account

### PR DESCRIPTION
**Pull request by duncanhall**
Ensures newly added accounts have the correct privileges to assume the given role.

### CODE REVIEW

File: server/commands/validators/awsAccountValidator.js [hunk 1]
```diff
@@ -26,10 +28,19 @@ function validate(account) {
...
     if (account.IsMaster && masterAccount !== undefined && account.AccountNumber !== masterAccount.AccountNumber)
-      throw new Error(`The account '${masterAccount.AccountName}' is already set as the master account.`)
+      throw new Error(`The account '${masterAccount.AccountName}' is already set as the master account.`);
+
+  }).then(co.wrap(function* () {
+    if (account.RoleArn !== null) {
+      try {
+        let roleIsValid = yield childAWSclient.assumeRole(account.RoleArn);
+      } catch (error) {
+        throw new Error(`${account.AccountName} is not permitted to assume the required role`);
```
Comment (Merlin-Taylor) [inline server/commands/validators/awsAccountValidator.js]: Possibly confusing: it's the master account that cannot assume the required role. `account.AccountName` refers to the name of the child account.

Comment (Merlin-Taylor) [inline server/commands/validators/awsAccountValidator.js]: May be worth logging the original error in the catch block so the stack trace isn't lost?

Comment (duncanhall) [inline server/commands/validators/awsAccountValidator.js]: Yep, agreed. Will update the message.

Comment (duncanhall) [inline server/commands/validators/awsAccountValidator.js]: Not sure we want the stack trace here? It's essentially a user error - they entered an incorrect role/ARN. Very little we can do about that with or without a stack trace.

**Pull request merged**